src/Orleans/Messaging/SocketManager.cs

```diff
@@ -65,9 +67,16 @@ private Socket SendingSocketCreator(IPEndPoint target)
                 s.NoDelay = true;
                 WriteConnectionPreemble(s, Constants.SiloDirectConnectionId); // Identifies this client as a direct silo-to-silo socket
                 // Start an asynch receive off of the socket to detect closure
-                var foo = new byte[4];
-                s.BeginReceive(foo, 0, 1, SocketFlags.None, ReceiveCallback,
```

Comment:
Not sure what this buys us, as this is just one place, and not even the primary one, where we use Socket.BeginReceive.  It's fine as a first step, if that's what we're after.